### ISSUE TITLE
Test Source Maps DebugID

### ISSUE THREAD

**Issue by krystofwoldrich** [getsentry/sentry-react-native#2892]
### Description

Verify that JS implementation of Debug ID works with RN. (If not, fix or implement for RN.)

Debug ID is used in[ Debug Meta Images ](https://develop.sentry.dev/sdk/event-payloads/debugmeta/#source-map-images)to identify the correct source map to use.

Related PR: [Remove `frames.abs_path`](https://github.com/getsentry/sentry-react-native/pull/2891), this is used to identify the files in JS, so likely this value will need to be adjusted for RN and added back. 

This enables users to upload source maps without specifying release.

**Comment by krystofwoldrich** [comment]
We need to inject the `Runtime Detection JS Script` into the JS bundle before it's processed by Hermes.

That means we need to inject it before we create the bundle as we can't inject command execution between `react-native bundle` and `hermesc -emit-binary`.

In the end, we need to inject the debug id into the generated source map.

For JSC the same steps would work.

At the moment the `inject command` only supports one path and expects both bundle and map to be there. That doesn't work for RN, as these files are typically in different dir structures.

```bash
# Current
sentry-cli sourcemaps inject <path>

# Option to make it compatible with RN
sentry-cli sourcemaps inject <path1> <path2> ... <pathN>
```

**Comment by kamilogorek** [comment]
Available in `2.15.1`.

**Comment by krystofwoldrich** [comment]
Injecting RN JSC bundles works. Hermes specific work will be tracked in https://github.com/getsentry/sentry-react-native/issues/2947.
